Add Dictionary class (#182)
spdx/spdx-3-model: model/Core/Classes/DictionaryEntry.md

SPDX-License-Identifier: Community-Spec-1.0

# DictionaryEntry

## Summary

A key with an associated value.

## Description

> The class used for implementing a generic string mapping (also known as associative arrays, dictionaries, and hash maps) in SPDX.  Each DictionaryEntry contains a key-value pair which maps the key to its associated value.  To implement a dictionary, this class is to be used in a collection with unique keys.

Reviewer: ```suggestion
The class used for implementing a generic string mapping (also known as associative array, dictionary, or hash map) in SPDX.  Each DictionaryEntry contains a key-value pair which maps the key to its associated value.  To implement a dictionary, this class is to be used in a collection with unique keys.
```
As "a generic string mapping" is singular, the following enumeration should be, too. The final "and" only makes sense to me if we know the list to be exhaustive. If that is the case, then ignore my "or" suggestion.

> The class used for implementing a generic string mapping (also known as associative arrays, dictionaries, and hash maps) in SPDX.  Each DictionaryEntry contains a key-value pair which maps the key to its associated value.  To implement a dictionary, this class is to be used in a collection with unique keys.

Reviewer: As an aside: I noticed you include two spaces after each period. Is there a reason for that?

Author: Habit.  Before [2020 when Microsoft declared 2 spaces](https://www.theverge.com/2020/4/24/21234170/microsoft-word-two-spaces-period-error-correction-great-space-debate) as incorrect, both single and double spaces were allowed.  In school, I (and millions of others) were taught that 2 spaces was correct.

Reviewer: Oh wow, I didn't know about that! :O

Reviewer 2: I'm a single-spacer myself[1], but the original reason for this  was that double-spaces emulated on the (fixed-width) typewriters the correct spacing that would be there with full movable type printing presses! Of course the only real solution in the digital era is to use TeX :grinning:

[1]: `M-: (setq sentence-end-double-space nil) RET`
